android_sms_gateway/client.py

```
@@ -124,31 +124,120 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
         return self._decrypt(
             domain.MessageState.from_dict(
-                self.http.get(f"{self.base_url}/message/{_id}", headers=self.headers)
+                self.http.get(f"{self.base_url}/messages/{_id}", headers=self.headers)
             )
         )
 
+    def get_messages(
+        self,
+        *,
+        filter: t.Optional[domain.MessagesQueryFilter] = None,
```

Ruff (0.15.10)

[error] Function argument `filter` is shadowing a Python builtin

(A002)